lib/ukallocbbuddy/bbuddy.c

```diff
@@ -231,13 +233,61 @@ static inline unsigned long num_pages_to_order(unsigned long num_pages)
 	return ukarch_flsl(num_pages - 1) + 1;
 }
 
+/*********************
+ * BALLOON SUPPORT
+ */
+
+static char using_balloon;
+
+/**
+ * Initializes the balloon once it is ready. Immediate for Xen,
+ * but later for KVM driver.
+ */
+static void balloon_init(struct uk_alloc *a)
+{
+	struct uk_bbpalloc *b;
+	size_t i;
+	int order = order;
+	chunk_head_t *chunk;
+	int r;
+
+	b = (struct uk_bbpalloc *)&a->priv;
+
+#ifdef CONFIG_PLAT_KVM
```

Comment:
I am really not fond of this differenciation. See other comments. I think that we need to come up with a better design here.